src/apps/intel_mp/intel_mp.lua

```diff
@@ -827,22 +827,36 @@ function Intel:sync_stats ()
    set(stats.speed, self:link_speed())
    set(stats.status, self:link_status() and 1 or 2)
    set(stats.promisc, self:promisc() and 1 or 2)
-   set(stats.rxbytes, self:rxbytes())
-   set(stats.rxpackets, self:rxpackets())
-   set(stats.rxmcast, self:rxmcast())
-   set(stats.rxbcast, self:rxbcast())
-   set(stats.rxdrop, self:rxdrop())
-   set(stats.rxerrors, self:rxerrors())
-   set(stats.txbytes, self:txbytes())
-   set(stats.txpackets, self:txpackets())
-   set(stats.txmcast, self:txmcast())
-   set(stats.txbcast, self:txbcast())
-   set(stats.txdrop, self:txdrop())
-   set(stats.txerrors, self:txerrors())
-   set(stats.rxdmapackets, self:rxdmapackets())
-   for idx = 1, #self.queue_stats, 2 do
-      local name, register = self.queue_stats[idx], self.queue_stats[idx+1]
-      set(stats[name], register())
+   -- Values only updated by master.
+   if self.master then
+      set(stats.rxbytes, self:rxbytes())
+      set(stats.rxpackets, self:rxpackets())
+      set(stats.rxmcast, self:rxmcast())
+      set(stats.rxbcast, self:rxbcast())
+      set(stats.rxdrop, self:rxdrop())
+      set(stats.rxerrors, self:rxerrors())
+      set(stats.txbytes, self:txbytes())
+      set(stats.txpackets, self:txpackets())
+      set(stats.txmcast, self:txmcast())
+      set(stats.txbcast, self:txbcast())
+      set(stats.txdrop, self:txdrop())
+      set(stats.txerrors, self:txerrors())
+      set(stats.rxdmapackets, self:rxdmapackets())
+   end
+   -- FIXME: rxcounter and txcounter are tightly coupled.
```

Comment:
I have a concern that we are changing the meaning of `run_stats` in a way that complicates the code and the use of the parameter. Specifying `run_stats` is supposed to allow the user to choose which instance updates the global counters, rather than whichever instance acquired the lock first. IIUC, this change makes this stop working because of the `if self.master then` check.

Could we instead make the instances sync their own counter no matter what the value of `run_stats` is? Then `init_queue_stats` would always be called (instead of guarded by `if self.run_stats then` in the constructor), and only add the queue stats that match the given `rxcounter` and `txcounter` would be added.

Then `sync_stats` would either always be called, with a guard specifying whether to update the global counters or it could be split into two methods (one that updates global counters, one that updates the per-queue counters).